serde/src/ser/mod.rs

```diff
@@ -111,13 +122,11 @@ pub trait Serializer {
     /// Serializes a `f64` value.
     fn serialize_f64(&mut self, v: f64) -> Result<(), Self::Error>;
 
-    /// Serializes a character. By default it serializes it as a `&str` containing a
-    /// single character.
+    /// Serializes a character. By default it serializes as bytes containing the UTF-8 encoding
+    /// of the character.
     #[inline]
     fn serialize_char(&mut self, v: char) -> Result<(), Self::Error> {
-        // FIXME: this allocation is required in order to be compatible with stable rust, which
-        // doesn't support encoding a `char` into a stack buffer.
-        self.serialize_str(&v.to_string())
+        self.serialize_bytes(::utils::encode_utf8(v).as_slice())
```

Comment:
@cmr does anything speak against changing this to `self.serialize_str(::utils::encode_utf8(v))`?